Lib/http/client.py

```diff
@@ -1126,7 +1126,11 @@ def putrequest(self, method, url, skip_host=False,
         url = url or '/'
         self._validate_path(url)
 
-        request = '%s %s %s' % (method, url, self._http_vsn_str)
+        if self._http_vsn_str:
+            request = '%s %s %s' % (method, url, self._http_vsn_str)
+        else:
+            # For HTTP 0.9 support
+            request = '%s %s' % (method, url)
```

Comment:
I believe interpolated strings are more readable than old style formatting (no need to constantly jump right and back while reading a line of code):

```suggestion
        if self._http_vsn_str:
            request = f'{method} {url} {self._http_vsn_str}'
        else:
            # For HTTP 0.9 support
            request = f'{method} {url}'
```